nemo/collections/speechlm2/models/duplex_ear_tts.py

```diff
@@ -0,0 +1,1991 @@
+        self.log("weights/RMS", weight_rms, on_epoch=True, sync_dist=True)
+        self.log("weights/max_abs", max_abs_w, on_epoch=True, sync_dist=True)
+        self.log("weights/mean", weight_mean, on_epoch=True, sync_dist=True)
+
+    def on_validation_epoch_start(self) -> None:
+        setup_audio_codec(self)
+        self.results_logger = ResultsLogger(self.validation_save_path).reset()
+        self.asr_bleu = ASRBLEU(self.cfg.scoring_asr).reset()
+        self.intelligibility = Intelligibility(self.cfg.scoring_asr, reuse_asr_hyps=True).reset()
+        self.secs = SECS(self.cfg.get("scoring_se", "titanet_large")).reset()
+
+    def on_validation_epoch_end(self, prefix="val") -> None:
+        asr_bleu = self.asr_bleu.compute()
+        for k, m in asr_bleu.items():
+            self.log(f"{prefix}_{k}", m.to(self.device), on_epoch=True, sync_dist=True)
+        cer_wer = self.intelligibility.compute()
+        for k, m in cer_wer.items():
+            self.log(f"{prefix}_{k}", m.to(self.device), on_epoch=True, sync_dist=True)
+        secs = self.secs.compute()
+        for k, m in secs.items():
+            self.log(f"{prefix}_{k}", m.to(self.device), on_epoch=True, sync_dist=True)
+
+    def get_teacher_force_inference_audio(self, batch, guidance_enabled=True):
+        inputs = self.prepare_inputs(batch)
+
+        tts_output = self.tts_model(
+            code=inputs["code"],
+            audio_mask=inputs["audio_mask"],
+            attention_mask=inputs["attention_mask"],
+            position_ids=inputs["position_ids"],
+            context_hidden_state=inputs["context_hidden_state"],
+            subword_ids=inputs["subword_ids"],
+            subword_mask=inputs["subword_mask"],
+            non_prompt_mask=inputs["non_prompt_mask"],
+            generation_config=self._get_generation_config(guidance_enabled=guidance_enabled),
+            teacher_forcing_inference=True,
+            guidance_enabled=guidance_enabled,
+        )
+        tf_audio_codes_pred = tts_output["codes"].squeeze(2)
+
+        # decode audio
+        tf_audio_codes_pred = replace_control_speech_codes(
+            tf_audio_codes_pred, self._control_codes, self.codec_silence_tokens
+        )
+        with ensures_target_precision(self.audio_codec_run_dtype), torch.no_grad():
+            audio_pred, audio_len = self.audio_codec.decode(tf_audio_codes_pred, inputs["output_lens"])
+
+        return audio_pred.squeeze(1), audio_len
+
+    def _get_generation_config(self, guidance_enabled: bool = False):
+        """Get default generation config for EAR-TTS."""
+        return {
+            "num_iter": 8,
+            "guidance_scale": self.cfg.get("inference_guidance_scale", 0.5) if guidance_enabled else None,
+            "top_p_or_k": self.cfg.get("inference_top_p_or_k", 0.8),
+            "noise_scale": self.cfg.get("inference_noise_scale", 0.8),
+            "eos_threshold": -3.0,
+        }
+
+    def offline_inference_with_custom_sentences(
```

Comment:
This should go into a dedicated inference script, and not into the public API of the model.
From a user's perspective - what would "custom sentences" even mean?